keps/sig-storage/3756-volume-reconstruction/README.md

```diff
@@ -0,0 +1,1030 @@
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+We have unit tests for the feature disabled or enabled.
+It affects only kubelet startup and we don't change format of data present in
+the OS (mount table, content of `/var/lib/kubelet/pods/`), so we don't have
+automated tests to start kubelet with the feature enabled and then disable it
+or a vice versa.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+If this feature is buggy, kubelet either does not come up at
+all (crashes, hangs) or does not unmount volumes that it should unmount.
+
+###### What specific metrics should inform a rollback?
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+`reconstructed_volumes_total`, `force_cleaned_failed_volumes_total`,
+`orphaned_volumes_cleanup_errors_total`
+
+See Observability in the detail design section. All newly introduced metrics
+will be added both to "old" and "new" VolumeManager, so users can compare
+these metrics with the feature gate enabled and disabled and see if downgrade
+actually helped.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+Not yet. This will be a manual test.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+
+No.
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+TODO whole chapter before GA.
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+###### How can someone using this feature know that it is working for their instance?
```

Comment:
Because this feature has no namespace user-facing knob, I think it's reasonable to indicate there is no way for a standard user to inspect one way or the other.

Reply:
They can read kubelet logs / metrics, if they have permissions to do so. Otherwise, no, there is no "regular user" facing knob.